File: problems/Java/19. Remove Nth Node From End of List.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode removeNthFromEnd(ListNode head, int n) {
        ListNode dumm = new ListNode(0, head);
        ListNode l = dumm;
        ListNode r = dumm;

        for (int i = 0; i < n; i++) {
            r = r.next;
        }

        while (r.next != null) {
            l = l.next;
            r = r.next;
        }

        l.next = l.next.next;
        return dumm.next;
    }
}

File: problems/Rust/19. Remove Nth Node From End of List.rs
#[derive(PartialEq, Eq, Clone, Debug)]
pub struct ListNode {
  pub val: i32,
  pub next: Option<Box<ListNode>>
}

impl ListNode {
  #[inline]
  fn new(val: i32) -> Self {
    ListNode {
      next: None,
      val
    }
  }
}

impl Solution {
    pub fn remove_nth_from_end(head: Option<Box<ListNode>>, n: i32) -> Option<Box<ListNode>> {
        let mut dumm = ListNode::new(0);
        dumm.next = head;
        let mut l = &mut dumm as *mut ListNode;
        let mut r = &dumm as *const ListNode;

        unsafe {
            for _ in 0..n {
                r = (*r).next.as_ref().unwrap();
            }

            while (*r).next.is_some() {
                l = (*l).next.as_mut().unwrap();
                r = (*r).next.as_ref().unwrap();
            }
            
            let temp = (*l).next.as_mut().unwrap().next.take();
            (*l).next = temp;
        }
        dumm.next
    }
}